docs/ref/contrib/admin/index.txt

```diff
@@ -2861,6 +2861,43 @@ Templates can override or extend base admin templates as described in
 ``AdminSite`` methods
 ---------------------
 
+.. method:: AdminSite.get_app_list(request, app_label=None)
+
+    .. versionadded:: 4.1
+
+    Returns a list of applications from the :doc:`application registry
+    </ref/applications/>` available for the current user. Each entry in the
+    list is a dict representing an application with the following keys:
+
+    * ``app_label``: the application label
+    * ``app_url``: the URL of the application index in the admin
+    * ``has_module_perms``: a boolean indicating if displaying and accessing of
+      the module's index page is permitted for the current user
+    * ``models``: a list of the models available in the application
+    * ``name``: name of the application
+
+    Each model is a dict with the following keys:
+
+    * ``model``: the model class
+    * ``object_name``: class name of the model
+    * ``name``: plural name of the model
+    * ``perms``: a ``dict`` tracking ``add``, ``change``, ``delete``, and
+      ``view`` permissions
+    * ``admin_url``: admin changelist URL for the model
+    * ``add_url``: admin URL to add a new model instance
+
+    List can be limited to a single application by providing ``app_label`` argument.
+
+    List is sorted alphabetically by application's ``name`` key. Additionally
+    ``models`` list of each application is sorted by model's ``name`` key. If
+    non-alphabetical order makes more sense and you want to customize application
+    and/or model order in admin site, you can override this method in a custom
+    AdminSite class.
+
+    .. warning::
+
+        Do not override AdminSite.get_app_list method for access control purposes.
+
```

Comment:
IMO this warning is unnecessary.

Reply:
Attack vector I had in mind was that somebody abused get_app_list to hide apps and models, but actually left admin pages pretty much accessible by direct links. I'm kinda paranoid about security, seen people do unbelievable crazy stuff :-(